state/executor.go

```diff
@@ -202,6 +211,35 @@ func (t *Transition) Receipts() []*types.Receipt {
 
 var emptyFrom = types.Address{}
 
+func (t *Transition) WriteFailedReceipt(txn *types.Transaction) error {
+	signer := crypto.NewSigner(t.config, uint64(t.r.config.ChainID))
+
+	var err error
+	if txn.From == emptyFrom {
+		// Decrypt the from address
+		txn.From, err = signer.Sender(txn)
+		if err != nil {
+			return NewTransitionApplicationError(err, false)
+		}
+	}
```

Comment:
You can use short-hand if statement for error checking.

Reply:
Fixed 1a5db97e2c54c08e84f7b6703d0b7a3c4f870e85